lib/jellyfish/application.ex

```diff
@@ -51,4 +51,50 @@ defmodule Jellyfish.Application do
     JellyfishWeb.Endpoint.config_change(changed, removed)
     :ok
   end
+
+  defp config_distribution(dist_config) do
+    ensure_epmd_started!()
+
+    # Release always starts in a distributed mode
+    # so we have to start a node only in development.
+    # See env.sh.eex for more information.
+    unless Node.alive?() do
+      case Node.start(dist_config[:node_name]) do
+        {:ok, _} ->
+          :ok
+
+        {:error, reason} ->
+          raise "Couldn't start Jellyfish node, reason: #{inspect(reason)}"
+      end
+
+      Node.set_cookie(dist_config[:cookie])
+    end
+
+    topologies = [
+      epmd_cluster: [
+        strategy: Cluster.Strategy.Epmd,
```

Comment:
Shouldn't cluster strategy be set up in the config instead of `application.ex`?

Reply:
I don't think so? What's the benefit?

Reply:
The main benefit I see is that we will have a clear separation of configuring and using configuration. Currently when someone wants to see how we use `libcluster` have to go to two files `application.ex` to see strategy and to `config_reader.ex` to see how hosts are represented. Also moving strategy to config will make it much easier for future extending to different strategies.

Reply:
I am not sure

> Currently when someone wants to see how we use libcluster have to go to two files application.ex to see strategy and to config_reader.ex to see how hosts are represented.

If we move to the config.exs we will still have two or even three places: configreader, config.exs, application.ex

>  Also moving strategy to config will make it much easier for future extending to different strategies.

That's a nitpick, I think that keeping everything in an application.ex is easier to read at this stage

Reply:
But we can move it to config_reader. The idea is to set configuration in one place e.g config_reader and in application.ex only read set values.

Reply:
Sorry but I am not convinced :stuck_out_tongue:   I see your point but I think it's better not to mix this config with env vars reading. At least for now

Reply:
As you wish, it's your responsibility as team leader, and you will have to carry the burden of this decision. 🤣